### DISCUSSION TITLE
Make vitest resolve sourcemap comments

### DISCUSSION THREAD

**Discussion by i-am-the-slime** [vitest-dev/vitest#3891] [Q&A] [answered]
I am running vitest on compiled PureScript code.
The way this works is that it takes `.purs` files in `src` and outputs files into:
`./output/Some.Module/index.js`.

When I enable sourcemaps in the PureScript compiler it will also produce the file:
`./output/Some.Module/index.js.map`

And it will append the following comment to the end of the `index.js` file mentioned above:

```
//# sourceMappingURL=index.js.map
```

Unfortunately, vitest still shows me this:

```
 ❯ output/Test.ExampleTest/index.js:58:31
     57|                       );
     58|                     });
     59|                   }
       |       ^
     60|                 );
     61|               });
```

Instead of resolving the issue back to the original `purs` file via that sourcemap comment and the corresponding `.map` file.

What am I doing wrong? How do I make this work?

**Comment by sheremet-va** [comment, marked as answer]
Vitest transforms all files into the format it can understand (replacing all imports with our own wrapper), so we have our own source map which is not merged with the already existing one.

Your use-case doesn't seem to be covered because Vitest assumes you use `plugins` to transform your files and didn't "pretransform" them before.

**Reply by i-am-the-slime** [reply]
Thanks for the answer. That's quite unfortunate. So in order to get this to work I'd have to write a transformer that'd call out to the compiler, right?

**Reply by sheremet-va** [reply]
Currently yes, this is an expectation. Probably should read the source map from the file, but this would be Vite responsibility I think.

**Reply by i-am-the-slime** [reply]
I'm trying this again today, curious if it's still the same today.

**Reply by hi-ogawa** [reply]
I think stacktrace should be able to point to `.purs` file:line:column properly, but Vitest error won't able to show code frame to display `.purs` code since that logic is currently assuming the file is loaded on js runtime.

If former is not even working, please feel free to provide a reproduction for us to take a look :pray:

**Comment by aleclarson** [comment]
I'd like to request an option for enabling sourcemap comments for specific globs. In other words, tell Vitest to read sourcemap comments for specific files and merge its own sourcemaps with those. Thanks!

**Reply by hi-ogawa** [reply]
Contrary to what's said in https://github.com/vitest-dev/vitest/discussions/3891#discussioncomment-6731744, Vite (and Vitest) should actually pick up existing `//# sourceMapping...` and merge it through Vite transform pipeline. Do you have an example which is not working?

We have a few tests cases, for example, added in https://github.com/vitest-dev/vitest/pull/7557. Note that this is only non-externalized dependencies. For externalized dependency, that's Node's territory and you probably need to add `--enable-source-maps` in execArgv https://vitest.dev/config/#pooloptions-forks-execargv
